ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandRunner.java

```diff
@@ -94,7 +107,9 @@ public CommandRunner(
       final String ksqlServiceId,
       final Duration commandRunnerHealthTimeout,
       final String metricsGroupPrefix,
-      final Deserializer<Command> commandDeserializer
+      final Deserializer<Command> commandDeserializer,
+      final CommandTopicBackup commandTopicBackup,
```

Comment:
passing the backup here leaks an implementation detail of the command topic (that we're using a backup to detect corruption/deletion). Instead, let's extend the CommandStore interface to include a `corruptionDetected` method that returns a boolean that we use here.

Reply:
done